khaos/khaos.go

```diff
@@ -76,14 +76,25 @@ func DrainNode(clientset *kubernetes.Clientset, khaosConfig *utils.Config) (err
 
   numNodes := len(nodes.Items)
   drainingNode := nodes.Items[rand.Intn(numNodes)]
+  if drainingNode.Spec.Unschedulable { // for now, we just skip if selected node is already cordoned
+    fmt.Printf("%s is already cordoned, skipping this event\n", drainingNode.Name)
+    return
+  }
+
   graceTime := int64(0)
 
   pods, err := podsInter.List(api.ListOptions{})
   fmt.Printf("Draining Node: %s\n", drainingNode.Name)
 
+  // make node unschedulable
+  drainingNode.Spec.Unschedulable = true;
+  _, err = clientset.Core().Nodes().Update(&drainingNode)
+  if err != nil { return err }
+
+  // kill pods on target node
   for _, pod := range pods.Items {
     if pod.Spec.NodeName == drainingNode.Name && pod.Name != khaosConfig.Name {
-      err = podsInter.Delete(pod.Name, &api.DeleteOptions{
+      err = clientset.Core().Pods(pod.ObjectMeta.Namespace).Delete(pod.Name, &api.DeleteOptions{
```

Comment:
This is actually a fix. The code in master actually doesn't work because deleting a pod requires a namespace name, and the `podsInter` used is created with `api.NamespaceAll` which is not valid for pod deletion. 

If the PR is not accepted, this fix still needs to be implemented. I can separate preemptively if necessary, I just made the fix here because I was already making changes.